python/ray/serve/_private/deployment_state.py

```diff
@@ -1551,18 +1519,29 @@ def deploy(self, deployment_info: DeploymentInfo) -> bool:
                 deployment_info.target_capacity,
             )
 
-        # Only allow the deployment status to become UPSCALING or DOWNSCALING
-        # if the deployment is not currently UPDATING.
-        allow_scaling_statuses = (
-            self.curr_status_info.status is not DeploymentStatus.UPDATING
-        )
+        old_target_state = self._target_state
+        self._set_target_state(deployment_info, target_num_replicas=target_num_replicas)
 
-        self._set_target_state(
-            deployment_info,
-            target_num_replicas=target_num_replicas,
-            status_trigger=DeploymentStatusTrigger.CONFIG_UPDATE_STARTED,
-            allow_scaling_statuses=allow_scaling_statuses,
-        )
+        # Determine if the updated target state simply scales the current state.
+        if self._target_state.is_scaled_copy_of(old_target_state):
+            old_num = old_target_state.target_num_replicas
+            new_num = self._target_state.target_num_replicas
+
+            if new_num > old_num:
+                self._curr_status_info = self._curr_status_info.handle_transition(
+                    DeploymentStatusInternalTrigger.MANUALLY_INCREASE_NUM_REPLICAS,
```

Comment:
Nitpick (non-blocker): Would be nice to use kwargs here. Same goes to few other places below where it's a mix of positional args and kwargs.